pkg/satellite/stream/stream.go

```diff
@@ -181,7 +182,36 @@ func (ss *satelliteStream) ackReceivedTelemetry(telemetryMessageAckId string) {
 	}
 }
 
+func (ss *satelliteStream) performAutoClose() {
+	log.Printf("Stream auto-close conditions met - exiting")
+	if ss.showStats {
+		metrics.logReport()
+	}
+	close(ss.closeTickerChan)
+	ss.Close()
+	os.Exit(0)
+}
+
 func (ss *satelliteStream) recvLoop() {
+	streamEndDetected := false
+
+	if ss.enableAutoClose {
+		ticker := time.NewTicker(1 * time.Second)
+		go func() {
```

Comment:
This goroutine calls `ss.performAutoClose()`, which closes `ss.closeTickerChan`, which triggers the `select` `case` in this goroutine, resulting in a bit of back-and-forth.

If we make the ticker a struct field, then it can be stopped from `ss.performAutoClose()`.

Would it not be even cleaner if we called `ss.performAutoClose()` from L330 and eliminate this goroutine altogether?